### ISSUE TITLE
Auth routes are not specified (documented, validated) 

### ISSUE THREAD

**Issue by illume** [kinvolk/nebraska#549]
## Auth routes are not specified (documented, validated)

Different auth backends have different inputs and responses.  They are chosen at startup time based on a cli option.

The current situation is that each auth backend uses the same URLs for routes. Also the inputs and responses are not specified/documented, and are manually validated (rather than using the backend generated validation).

See:
- spec.yaml /login/cb
- backend/pkg/auth/oidc.go@ValidateToken
- backend/pkg/auth/oidc.go@LoginToken
- previous conversation in code review at https://github.com/kinvolk/nebraska/pull/513#discussion_r754099574

## Impact

Understanding the code is more difficult(for backend developers, API users, and operators). Maintenance is harder, because changes in the spec need to be done and kept in sync manually. Operators don't have a documented REST API spec, meaning they might find it difficult to know how it works and complete tasks like hardening the API. Validation can not be done by automatically by the backend or the frontend (currently only the backend is using the spec for automatic validation).

Since this area of code has no tests, not having docs or a spec here means the code is lower quality. This is not an ideal situation for portions of the codebase which have a high impact on the security of the system (as auth does).

**Comment by tylerauerbeck** [comment]
@ervcz For this one (and maybe just in general), would it maybe be beneficial to just better expose the openapi spec found at `backend/api/spec.yml` ? Maybe just adding a swagger/openapi endpoint so that folks can better explore it?

**Comment by tylerauerbeck** [comment]
@ervcz I was thinking something interactive may help here? I've got this working at the moment with redoc (although the [library](https://github.com/kohkimakimoto/echo-openapidocs) I'm using has a number of different generators you can use).

The only thing I have to figure out is how to get this to play nicely with [`oapirequestvalidator`](https://github.com/flatcar/nebraska/blob/main/backend/pkg/server/server.go#L109) because it only works when I've got that commented out at the moment.

Before I had too much farther down this path, let me know what you think.

<img width="1454" alt="Image" src="https://github.com/user-attachments/assets/1fd2d5e8-042a-4d77-9c4c-dd7019748a97" />

**Comment by tylerauerbeck** [comment]
I agree keeping the number of dependencies small and low here is beneficial from a maintainer perspective. 

The reason I'm thinking an interactive component here would be beneficial would be from the developer/user experience side of things:
- allows a user/developer to interact specifically with the API that they're working against. If we make the user go search out the  spec.yaml, they're likely to grab the wrong version (i.e. just grabbing what's available in `main` instead of the version they're working against
- Having users need to use an upstream online spec doesn't necessarily add a dependency to the project, but it adds a developer dependency that is then out of our control. Not to mention that there are likely going to be restrictions from implementation to implementation where users may not actually have access to somewhere that we recommend.
- In practice, having the user have to navigate away from the project reduces the likelihood of them returning to either a) make use of the project or b) continue on in a potential contribution that they may be making (attention spans are short these days 😬 )

By adding a dependency though, what we gain is the ability to provide users with the space to interact with the API and discover what they need while remaining engaged within the project and allow them to understand what's available within the version that they're working in without putting any blockers at their path simply by hitting something like `https://nebraska.example.com/swagger`
